index.d.ts

```diff
@@ -1,7 +1,203 @@
 // Clear all type of caches in Skia
 export function clearAllCache(): void
 
-export interface DOMMatrix2DInit {
+export type GlobalCompositeOperation =
+  | 'clear'
+  | 'copy'
+  | 'destination'
+  | 'source-over'
+  | 'destination-over'
+  | 'source-in'
+  | 'destination-in'
+  | 'source-out'
+  | 'destination-out'
+  | 'source-atop'
+  | 'destination-atop'
+  | 'xor'
+  | 'lighter'
+  | 'normal'
+  | 'multiply'
+  | 'screen'
+  | 'overlay'
+  | 'darken'
+  | 'lighten'
+  | 'color-dodge'
+  | 'color-burn'
+  | 'hard-light'
+  | 'soft-light'
+  | 'difference'
+  | 'exclusion'
+  | 'hue'
+  | 'saturation'
+  | 'color'
+  | 'luminosity'
+  | 'saturate'
+
+export type CanvasLineCap = 'butt' | 'round' | 'square'
+
+export type CanvasLineJoin = 'bevel' | 'miter' | 'round'
+
+export type CanvasTextBaseline = 'alphabetic' | 'bottom' | 'hanging' | 'ideographic' | 'middle' | 'top'
+
+export type CanvasTextAlign = 'center' | 'end' | 'left' | 'right' | 'start'
+
+export interface TextMetrics {
+  readonly actualBoundingBoxAscent: number
+  readonly actualBoundingBoxDescent: number
+  readonly actualBoundingBoxLeft: number
+  readonly actualBoundingBoxRight: number
+  readonly fontBoundingBoxAscent: number
+  readonly fontBoundingBoxDescent: number
+  readonly width: number
+}
+
+export type CanvasFillRule = 'evenodd' | 'nonzero'
+
+export class CanvasRenderingContext2D {
+  drawImage(image: Canvas | Image, dx: number, dy: number): void
+  drawImage(image: Canvas | Image, dx: number, dy: number, dw: number, dh: number): void
+  drawImage(
+    image: Canvas | Image,
+    sx: number,
+    sy: number,
+    sw: number,
+    sh: number,
+    dx: number,
+    dy: number,
+    dw: number,
+    dh: number,
+  ): void
+  putImageData(imagedata: ImageData, dx: number, dy: number): void
+  putImageData(
+    imagedata: ImageData,
+    dx: number,
+    dy: number,
+    dirtyX: number,
+    dirtyY: number,
+    dirtyWidth: number,
+    dirtyHeight: number,
+  ): void
+  getImageData(sx: number, sy: number, sw: number, sh: number): ImageData
+  createImageData(sw: number, sh: number): ImageData
+  createImageData(imagedata: ImageData): ImageData
+  /**
+   * For PDF canvases, adds another page. If width and/or height are omitted,
+   * the canvas's initial size is used.
+   */
+  addPage(width?: number, height?: number): void
```

Comment:
PDF is not supported in this library though 🤔

Reply:
#309

Reply:
Apologies I should have clarified this.

The following types were directly copied from `skia-canvas` which this project depends on:
```
GlobalCompositeOperation
CanvasLineCap
CanvasLineJoin
CanvasTextBaseline
CanvasTextAlign
TextMetrics
CanvasFillRule
CanvasRenderingContext2D
CanvasGradient
CanvasPattern
```
It is correct to be bringing in these types, but as you rightfully point out, this library does not currently implement all features. I was reluctant to start removing type properties / class methods as I'm uncertain which are, and are not, implemented. I'm hoping someone with more experience (i.e. the maintainers) can help out here. 

The only modifications I've made are to `CanvasRenderingContext2D`:
1. To add method `setTransform(a: number, b: number, c: number, d: number, e: number, f: number): void` in addition to the existing `setTransform(transform?: DOMMatrix2D): void`
2. Modify `fill(path?: Path2D, fillRule?: CanvasFillRule): void` to `fill(fillRule?: CanvasFillRule): void`

These changes now allow my project to build (`tsc`) where before it was failing (as described in #659).